server/src/auth/auth.service.ts

```diff
@@ -16,14 +22,68 @@ export class AuthService {
 
   /** Get identifier from a google token */
   getIdFromGoogleToken(token: string): string | undefined {
-    return undefined;
+
+    return undefined
+  }
+
+  
+  async verifyGoogle(token_id : string): Promise<any> {
+    try {
+      const client = new OAuth2Client(AuthConstants.client_id);
+      const ticket: any = await client.verifyIdToken({
+        idToken: token_id,
+        audience: AuthConstants.client_id, 
+      });
+      const payload = ticket.getPayload();
+      //const userid = payload['sub'];
+      return payload
+    }
+    catch(error) {
+      if (error) {
+        return "error"
+      }
+    }
+    return "error"
   }
 
+
+  async loginGoogle(ID_token : string) {
+    const payload = await this.verifyGoogle(ID_token);
```

Comment:
You can factor out the stuff in this method to use more than just Google. If you skip the verifyGoogle part and have the method take in AuthType, you can rename this method to "login" so you won't have to repeat the code.